packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -61,21 +60,40 @@ const getFunctionMetadataMap = async (
 async function generateFunctionMetadata(
   filepath: string
 ): Promise<[string, FunctionMetadata]> {
-  const buildResult = await esbuild.build({
-    entryPoints: [filepath],
-    outdir: TEMP_DIR,
-    write: false,
-    format: "esm",
-    bundle: true,
-  });
-  const importedFile = await importFromString(buildResult.outputFiles[0].text);
+  project.addSourceFileAtPath(filepath);
```

Comment:
Is there a way to point to the functions dir when initializing the ts-morph project instead of adding each file manually? IIRC, each time you add a file to the project like this, ts-morph recreates the whole tsc instance, which impacts performance

Reply:
cool! that made it 2x faster!

Reply:
I did it a little differently-- instead of adding it all on project instantiation, I just add all the files in one function call.  If we wanted to do it on project instantiation, we would have to make a tsconfig file specifying the directory we wanted, and pass the file path to the function.  I think that having a random floating tsconfig file just for the `functions` dir would be weird, so I went with this approach.